greedy_algorithms/gale_shapley.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file
+ * @brief Stable Marriage Problem implementation
+ * @details
+ * This implementation utilizes the Gale-Shapley algorithm to find stable matches
+ * between men and women based on their preferences.
+ *
+ * **Stable Marriage Problem** aims to find a stable matching between two equally sized 
+ * sets of elements given an ordinal preference for each element. The algorithm was
+ * introduced by David Gale and Lloyd Shapley in 1962.
+ *
+ * @author [B Karthik](https://github.com/BKarthik7)
+ */
+
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <algorithm>
+using namespace std;
+
```

Comment:
```suggestion
namespace greedy_algorithms {
```
this algorithm should be inside the greedy algorithms namespace

Reply:
can you explain this i didnot understood